design-documents/services-decomposition-guidelines.md

```diff
@@ -0,0 +1,35 @@
+# Services decomposition guidelines
+
+## About
+
+Document contains general recommendations and principles that need be considered and followed during services decomposition.
+
+## Guidelines
+
+Services should have it's own core_config_data table that would contain settings specific to the service. Services should expose API to modify settings in core_config_data. Admin panel should support editing settings in the local database (settings for BFF) and for individual services via remote calls.
+
+Services should not have knowledge about stores/websites. As some services store information about stores/websites with it's data stores/websites may need to be validated. If validation is critical services can request information about stores/website from websites service to use for validation. If not, validation can be omitted.
+
+Modules use directory data (countries, cities, states). This data need to be reused between different services. As it almost never changes, it doesn't make much sense to add directory service. Directory data can be moved to configuration, so it can be shared across different services.
+
+Modules have dependency on session. Handling session should be responsibility of BFF. BFF can pass user data to the service if needed.
+
+For reporting purposes, service should send data to reporting service (BFF after first iteration). Can be done by cron, real time depending on data and types of reports.
```

Comment:
I believe inverted solution should be used.  Reporting system should subscribe on data changes and create aggregates used in report

Reply:
Agree, changed.